setup.cfg

```diff
@@ -48,8 +48,8 @@ package_dir=
     =src
 packages=find_namespace:
 install_requires=
-    opentelemetry-api~=1.12
-    opentelemetry-sdk~=1.12
+    opentelemetry-api~=1.14
+    opentelemetry-sdk~=1.14
```

Comment:
I think the old version range should already allow for `>= 1.12,  ==1.*` ([see definition](https://peps.python.org/pep-0440/#compatible-release)), there should be no change necessary. :slightly_smiling_face:

It will only pull in the `1.12` dependencies when there is no API/SDK installed yet.

Reply:
Hmm I wonder if we should actually specify a _minimum_ version instead of the current version

Reply:
If we keep this change (1.14) I guess we might also want to update the readme file, where we state it was built "against 1.12"? Not a blocking thing anyway.

Reply:
I was wrong with my comment above, it only pulled in `1.12` because I had that package cached in `pip`.
`~=1.12` will actually pull in the current version even though it was not installed in my `venv`.

We could change this to `>=1.12, <2` and it should not change anything semantically. 
`=~1.12`,`>=1.12,<2`, `>=1.12,==1.*`, and `>=1.12,<2` should all be semantically equivalent.